sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1159,7 +1159,7 @@ case class StringLocate(substr: Expression, str: Expression, start: Expression)
   override def children: Seq[Expression] = substr :: str :: start :: Nil
   override def nullable: Boolean = substr.nullable || str.nullable
   override def dataType: DataType = IntegerType
-  override def inputTypes: Seq[DataType] = Seq(StringType, StringType, IntegerType)
+  override def inputTypes: Seq[DataType] = Seq(StringType, IntegerType)
```

Comment:
the expected input types here should match child expressions. The i-th position in the returned seq indicates the type requirement for the i-th child.

So I guess this change is wrong.

Reply:
Please ignore me~